internal/defaults.go

```diff
@@ -0,0 +1,8 @@
+package internal
+
+import "time"
+
+const (
+	DefaultConnectionTimeoutWithoutFailover = 24 * 366 * time.Hour
```

Comment:
Why is this different than the value that I see for the max duration ?  
(1<<63 - 1)
This is a year right , why don't we use infinite ?

Reply:
The timeout is added to the current time to create the deadline. If we use max duration, then it would overflow. Isn't 1 year equivalent to infinity?

Reply:
I think we can easily fix the code to avoid addition so that we can get rid of confusing magic numbers